libs/wingc/src/type_check.rs

```diff
@@ -724,6 +724,14 @@ impl TypeRef {
 		}
 	}
 
+	pub fn is_iterable(&self) -> bool {
+		if let Type::Array(_) | Type::Set(_) | Type::MutArray(_) | Type::MutSet(_) = **self {
```

Comment:
@hasanaburayyan - maybe not part of this PR, but what about `Json`? Maybe open an issue to support iterating over a json-array?